woocommerce/woocommerce-ios
# [Subscriptions] Sync subscriptions when order details is loaded

Part of #8950
⚠️ Depends on #9531


## Description

This PR adds support for syncing subscriptions when the order detail screen is loaded for an order:

* It waits for the order to be synced, to ensure the `Order.renewalSubscriptionID` property is available (if it is a subscription renewal order).
* It checks for the Subscriptions extension and only syncs subscriptions if that plugin is active.
* If there are available subscriptions, it loads them into `OrderDetailsDataSource`.

The UI changes (to display the synced subscriptions) are in a separate PR.

## Testing instructions

The synced subscriptions are not yet used in the UI. You can check that they are synced correctly with the following steps.

### Prerequisites

1. Install and activate the [Subscriptions extension](https://woocommerce.com/products/woocommerce-subscriptions/) on your store.
2. Create at least one subscription product.
3. Create an order for that subscription product, which creates a new subscription.

### To test

1. Add a breakpoint to `OrderDetailsViewModel` on L611.
2. Build and run the app.
3. Go to the Orders tab.
4. Open the subscription order and confirm you hit the breakpoint and your order's subscription is in the `result` being added to `OrderDetailsDataSource`.

---
